crates/iceberg/src/io/file_io.rs

```diff
@@ -95,9 +95,14 @@ impl FileIO {
     /// # Arguments
     ///
     /// * path: It should be *absolute* path starting with scheme string used to construct [`FileIO`].
-    pub async fn remove_all(&self, path: impl AsRef<str>) -> Result<()> {
+    pub async fn remove_dir_all(&self, path: impl AsRef<str>) -> Result<()> {
```

Comment:
is this a breaking change? i.e. something we should warn users about on the next release